zip-0226.rst

```diff
@@ -1,8 +1,313 @@
+
+This, however, brings some issues when it comes to adding multiple Asset identifiers, as the output note of the split Actions *cannot* be of *any* Asset identifier, it must be enforced to be an actual output of a GroupHash computation (in fact we want it to be of the same identifier as the original input note, but the binding signature takes care that the proper balancing is performed). If not, then the prover could essentially input a multiple (or linear combination) of an existing identifier, with the goal to attack the network by overflowing the ZEC value balance and hence counterfeiting ZEC funds.
+
+In order to prevent this, we make some modifications to the circuit. Specifically we remove the dummy note functionality for custom Assets and we enforce that *every* input note to an ZSA Action must be proven to exist in the set of note commitments in the note commitment tree. We then enforce this real note to be “unspendable” in the sense that its value
+will be zeroed in split Actions and the nullifier will be randomized, making the note not spendable in the specific Action. Then, the proof itself ensures that the output note is of the same identifier as the input note. In the circuit, the split note functionality will be activated by a boolean private input to the proof.
+
+This creates a chain of induction that ensures that the value base points of all output notes of a transfer are actual outputs of a GroupHash, as they originate in the Issuance protocol which is publicly verified. If this were not done then it would be possible to violate balance, for example by using a value base point derived from those of other Assets.
+
+Note that we do not care about whether the note identified by a Split Input is owned by the sender, or whether it was nullified before.
+
+Wallets and other clients have a choice to make to ensure the Asset type is preserved for the output note of a Split Action:
+
+1. The Split Input note could be another note of the same Asset identifier that is being spent by this transaction (but not by this Split Input). 
+2. The Split Input note could be a different unspent note of the same :math:`\mathsf{AssetId}` (note that the note will not actually be spent).
+3. The Split Input note could be an already spent note of the same :math:`\mathsf{AssetId}` (note that by zeroing the value in the circuit, we prevent double spending).
+
+The specific circuit changes are presented below.
+
+Circuit Statement
+=================
+
+The advantage of the design described above, with respect to the circuit statement, is that every *ZSA Action statement* is kept closely similar to the Orchard Action statement [#protocol-actionstatement]_, except for a few additions that ensure the security of the Asset identifier system.
+
```

Comment:
Replace with a sentence that describes what this section is specifying: the new circuit, or a modification to the NU5 Orchard Action statement?

Reply:
Done.